wp-content/plugins/core/src/Nav_Menus/Nav_Attribute_Filters.php

```diff
@@ -11,6 +11,61 @@
  */
 class Nav_Attribute_Filters {
 
+	/**
+	 * Generate a class prefix based on wp_nav_menu() parameters.
+	 *
+	 * Because `menu` overrides `theme_location`, and `theme_location` is used if
+	 * `menu` is empty or menu does not exist, the solution is to get the slugs
+	 * of both menus, and if they're equal, use the theme location. If they're not equal,
+	 * then use the menu slug.
+	 *
+	 * @param stdClass $args An object of {@see wp_nav_menu()} arguments.
+	 *
+	 * @return string
+	 *
+	 * @todo evaluate: prepending "location-" and "menu-" is backwards-compatibility issue
+	 */
+	protected function get_class_prefix( stdClass $args ): string {
+		$menu = wp_get_nav_menu_object( $args->menu );
+		$locations = get_nav_menu_locations();
+
+		$menu_slug = '';
+		$location_menu_slug = '';
+
+		// Get slug of menu if exists.
+		if ( $menu ) {
```

Comment:
`wp_get_nav_menu_object()` is filterable, I'd check directly that `$menu->slug` itself exists, and as a suggestion you could combine that with the initial define which improves readability. 

At some point scrolling through this large code block you have to back up and check if things have been initialized or not.

e.g.

```php
$menu_slug = ( $menu->slug  ?? '' ) ? $menu->slug : '';
```

Reply:
> check directly that $menu->slug itself exists

👍 

> At some point scrolling through this large code block you have to back up and check if things have been initialized or not.

I very rarely use ternary operators, as it decreases readability, especially for junior devs. Remembering the variables were initialized or scrolling up a little bit to confirm is much easier. ✌️